Assets/MixedRealityToolkit.Tests/TestUtilities.cs

```diff
@@ -41,18 +41,6 @@ public static void CleanupScene()
                 return;
             }
 #endif
-
-            if (testScene.IsValid() && testScene.isLoaded)
-            {
-                SceneManager.UnloadSceneAsync(testScene);
-            }
-
-            testScene = SceneManager.CreateScene("TestScene");
```

Comment:
Is this not needed? The comment above mentions that the remaining lines work in edit mode (and are blocked by a `!EditorApplication.isPlaying` check), but that we need to set the scene up manually in play mode.

Reply:
I saw those comments too, and I didn't initially have it. However, when I had the code above I noticed that when I ran all new 3 playmode tests that I had, the first test passed, but then the last few tests would fail (I forget which error now).

After debugging for a bit I realized that when we create a new scene for every tests, the following happens:

1. We create a scene, with one version of MRTK A for scene A.
2. We unload / remove scene A, but MRTK for scene A is still around.
3. We create a new scene B, but the MRTK singleton that we are using is still from Scene A.
4. As we use scene B, but MRTK doesn't get any events from scene B because it seems to still be tied to scene A.

When I removed the code to actually create this test scene, I actually managed to get all tests to pass. The reason I think this happens is that each test class creates and then properly tears down its own scene, including tearing down MRTK properly. So we get MRTK A with Scene A, MRTK B with Scene B.

Reply:
Okay, I am wrong. Here is what I think happens in Play Mode Tests:

1. At start of test run, test scene is created, with MRTK A
2. All tests run in that one scene. 

If we want to ensure that for each class / each test, we use a new scene, we need to destroy / create the scene in cleanup. Only problem is that when you create a new scene, the old MRTK instance is still sticking around. So if you create Scene B, the MRTK from Scene A is still around, it seems.

I see two approaches:

1. Destroy the scene for each test.
2. In the cleanup for each scene, destroy all gameObjects in the scene. Could be very expensive.

Reply:
Upon even further investigation (I tried adding back the technique of unloading old scene, and creating new one), I have realized why I removed the code in the first place. Two reasons:

First, when I initially tried using this code and running two tests, I ran into the following error: "A scene named 'TestScene' already exists". After I resolved that error by waiting for the async op to finish before loading a new scene, I ran into my second issue.

The second issue is what I mentioned above: that the MRTK from the old scene was still sticking around. The main symptom of this was that MRTK was throwing errors saying a main camera existed when it didn't. Also, after unloading a scene I would check in debugger in MRTK was still initialized, and it was.

To avoid going down a rabbit hole, I think for now a good policy is just to have all playmode tests run in the same scene. This currently means that all tests need to make sure to remove all objects they add. It's not great and I think we should solve the problem that MRTK is still around even when a scene is destroyed. However, I don't want to block completing this PR to do that, since this PR fixes bugs to manipulationhandler and does add new tests.

@Railboy and @lukastoenneMS  I'd love to hear your thoughts on this.

Reply:
I think running on the same scene for these particular tests is fine.

Once we actually make runtime modifications to scenes it becomes more difficult, because to reliably run the tests we'd want to start from a clean slate and (re)load a new scene.